### ISSUE TITLE
Multiple primary key defined

### ISSUE THREAD

**Issue by shengxq** [typeorm/typeorm#1652]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `websql`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

Using typeorm with JavaScript,
entity use json

```
{
    "name": "huiyuan",
    "columns": {
        "gsbh": {
            "primary": true,
            "type": "int"
        },
        "hybh": {
            "primary": true,
            "type": "int",
            "generated":"increment"
        }
        ,
        "name": {
            "type": "varchar"
        }
    }
}
```

The output SQL:

> CREATE TABLE `huiyuan` (`gsbh` int(11) NOT NULL, `hybh` int(11) NOT NULL PRIMARY KEY AUTO_INCREMENT, `name` varchar(255) NOT NULL, PRIMARY KEY(`gsbh`)) ENGINE=InnoDB 

The error message:
Error: ER_MULTIPLE_PRI_KEY: Multiple primary key defined

The following is OK:
================================
```
{
    "name": "huiyuan",
    "columns": {
        "gsbh": {
            "primary": true,
            "type": "int"
        },
        "hybh": {
            "primary": true,
            "type": "int"
        }
        ,
        "name": {
            "type": "varchar"
        }
    }
}
```

> CREATE TABLE `huiyuan` (`gsbh` int(11) NOT NULL, `hybh` int(11) NOT NULL, `name` varchar(255) NOT NULL, PRIMARY KEY(`gsbh`, `hybh`)) ENGINE=InnoDB

**Comment by pleerock** [comment]
your issue is empty.

**Comment by shengxq** [comment]
修改好了
It has been modified.

**Comment by pleerock** [comment]
You cannot have generated and non-generated primary in a single table, consider removing generated from your composite primary or left only one generated primary.

**Comment by shengxq** [comment]
This allows you to create a table:
================
CREATE TABLE `huiyuan` (
  `gsbh` int(11) NOT NULL AUTO_INCREMENT,
  `hybh` int(11) NOT NULL,
  `name` varchar(255) NOT NULL,
  PRIMARY KEY (`gsbh`,`hybh`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

**Comment by shengxq** [comment]
mysql ver  5.7

**Comment by pleerock** [comment]
yes but its only surface of iceberg. There are lot of issues if you'll need to modify such table. We'll think if we can implement this functionality and resolve all issues in next version.

**Comment by shengxq** [comment]
I just found out to tell you that I haven't decided to use it in the project.
thanks

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.25`. You can try it via `npm i typeorm@next`.
